bigday/settings.py

```diff
@@ -17,14 +17,24 @@
 
 
 # Quick-start development settings - unsuitable for production
-# See https://docs.djangoproject.com/en/1.9/howto/deployment/checklist/
+# See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'u7!-y4k1c6b44q507nr_l+c^12o7ur++cpzyn!$65w^!gum@h%'
+# This is a default value and must be changed!
+# Rename "localsettings.py.template" to 'localsettings.py' and edit your settings.
+# To protect your credentials from leaking to your Git server we added 'localsettings.py' to the gitignore
+try:
+    from .localsettings import ownSecretKey
+    SECRET_KEY = ownSecretKey.OWN_SECRET
```

Comment:
is there a reason you chose to namespace all of these?

It's more common to define the defaults in settings.py and then have:

```python
from .localsettings import *
````

somewhere. (There are other ways of doing this, but this is the most common for using a `localsettings` pattern). In that scenario you just use the same variable names and put them directly in the file. So in localsettings you'd have:

```python
SECRET_KEY = '***'
```

I think grouping all the mail settings together makes some sense, but it might be simpler not to bother with the others (especially this one). Also it's a bit odd to me that they are renamed (I didn't notice this with the mail settings).

Having given it some thought, I think I'd prefer to remove the namespacing and use the same settings names, since this is a more common pattern in Django that will be more familiar to people.

Reply:
I was taught, that explicit is better than implicit and `import *` is bad style.
According to https://peps.python.org/pep-0020/ and https://stackoverflow.com/questions/2386714/why-is-import-bad

I see, why this is maybe overkill in this case.

So, if I import the whole file, the variables are automatically set and I just need to set them in the `except` statement like:

```Python
try:
  from .localsettings import *
except:
  DEFAULT_WEDDING_FROM_EMAIL = 'something'
```

I can surely change it, give me a quick ping, if the "try import all and set stuff in the except clause" works for you.

Reply:
I think it's better to define the defaults above the `try`. Otherwise I think it's not possible to only override a small subset of them?

And good point re: `import *`. IMO this is one of the rare cases where it's justified, since you're actually just trying to bring it whatever is in that file into another file.

Reply:
Will there be a case, were we need to just for example need to set the couple data and not the sender address?

I will edit the PR in the evening to the "set everything first and then do a full overwrite with the import" paradigma :).

Reply:
Awesome, thanks!

> Will there be a case, were we need to just for example need to set the couple data and not the sender address?

Some people might prefer to edit non-secrets directly in `settings.py`.

Reply:
I think the whole discussion sits on a bad practice. "local_settings" or similar "change on your machine but don't commit" methods were encouraged by people in the early days of Django but for a while, using the env is the go to solution on that end.

Reply:
Also, I would prefer to separate these setting management-related changes from the actual scope of the PR. Sending refactoring and feature fixes/improvements in a single PR makes it hard to read and understand in the future.

Reply:
I don't disagree with you on using env, though this is still an improvement on the current situation.

Given that this repo is very inactive (it is usually forked and never merged back) I'm happy to take the improvements we can get!

That said - if you want to submit a PR moving everything to `django-environ` or what not, I'd happily merge that too...

Reply:
> I think the whole discussion sits on bad practice. "local_settings" or similar "change on your machine but don't commit" methods were encouraged by people in the early days of Django but for a while, using the env is the go-to solution on that end.

I'm not used to Django, so that was the first thing, which went on my mind :). Maybe I'll read up the "new way" in the future. 
Thanks for the feedback!